test/unit/services/query-service.test.js

```diff
@@ -119,5 +119,219 @@ describe('query-service', () => {
       expect(validatedResponseNoContext).toStrictEqual(false)
       expect(validatedResponseNoAnswer).toStrictEqual(false)
     })
+
+    test('fetchAnswer returns summary answer when summariesEnabled is true', async () => {
+      createRetrievalChain.mockResolvedValue({
+        invoke: jest.fn().mockResolvedValue({
+          answer: JSON.stringify({ answer: 'summary response', items: [] })
+        })
+      })
+
+      const prompt = jest.fn()
+      const input = 'deer fencing'
+      ChatPromptTemplate.fromMessages.mockReturnValue(prompt)
+
+      const { response, summariesMode, hallucinated } = await fetchAnswer({}, input, [], false, true)
+
+      expect(JSON.parse(response).answer).toStrictEqual('summary response')
+      expect(createStuffDocumentsChain).toHaveBeenCalledWith(
+        expect.objectContaining({ prompt })
+      )
+
+      if (!hallucinated) {
```

Comment:
A conditional expect feels like a code smell. Why do we need the if statement check?

Reply:
because in some cases it might hallucinate on summaries and default to regular search
but not vice versa

Reply:
Shouldn't the response always be the same based on the mocked AI search and OpenAI?